pkg/views/gitprovider/select.go

```diff
@@ -109,32 +111,46 @@ func GitProviderSelectionView(ctx context.Context, gitProviderAddView *apiclient
 	).WithTheme(views.GetCustomTheme())
 
 	views.RenderInfoMessage(getGitProviderHelpMessage(gitProviderAddView.ProviderId))
-	err = userDataForm.Run()
+	err := userDataForm.Run()
 	if err != nil {
-		log.Fatal(err)
+		return err
 	}
 
+	return nil
 }
 
-func GitProviderDeleteView(gitProviderAddView *apiclient.SetGitProviderConfig, userGitProviders []apiclient.GitProvider, apiClient *apiclient.APIClient, ctx context.Context) {
+func GetGitProviderFromPrompt(ctx context.Context, gitProviders []apiclient.GitProvider, apiClient *apiclient.APIClient) (*apiclient.GitProvider, error) {
 	var gitProviderOptions []huh.Option[string]
-	for _, userProvider := range userGitProviders {
-		gitProviderOptions = append(gitProviderOptions, huh.Option[string]{Key: fmt.Sprintf("%s   %s", userProvider.ProviderId, userProvider.Alias), Value: userProvider.Id})
+	for _, userProvider := range gitProviders {
+		gitProviderOptions = append(gitProviderOptions, huh.Option[string]{
+			Key:   fmt.Sprintf("%s   %s", userProvider.ProviderId, userProvider.Alias),
+			Value: userProvider.Id,
+		})
 	}
+
+	var selectedId string
 	gitProviderForm := huh.NewForm(
 		huh.NewGroup(
 			huh.NewSelect[string]().
 				Title("Choose a Git provider").
 				Options(
 					gitProviderOptions...,
 				).
-				Value(gitProviderAddView.Id)).WithHeight(8),
+				Value(&selectedId)).WithHeight(8),
 	).WithTheme(views.GetCustomTheme())
+
 	err := gitProviderForm.Run()
 	if err != nil {
-		log.Fatal(err)
+		return nil, err
+	}
+
+	for _, userProvider := range gitProviders {
```

Comment:
To eliminate this and make the code more readable, we can have the huh select have the object as a value instead of the ID.